src/components/Tokens/TokenDetails/TokenDetail.tsx

```diff
@@ -174,6 +169,98 @@ const MissingData = styled.div`
   gap: 20px;
 `
 
+export const AboutContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 16px;
+  padding: 24px 0px;
+`
+
+const TokenDescriptionContainer = styled.div`
+  overflow: hidden;
+  text-overflow: ellipsis;
+  max-width: 100%;
+  max-height: fit-content;
+`
+
+const TruncateDescriptionButton = styled.div`
+  color: ${({ theme }) => theme.textSecondary};
+  font-weight: 400;
+  font-size: 14px;
+  flex: none;
+  padding: 12px 0px;
+
+  &:hover,
+  &:focus {
+    color: ${({ theme }) => darken(0.1, theme.textSecondary)};
+    cursor: pointer;
+  }
+`
+
+type TokenDetailDataProps = {
+  description: string | null
+  homepageUrl: string | null
+  twitterName: string | null
+}
+
+export function AboutSection({ address, tokenDetailData }: { address: string; tokenDetailData: TokenDetailDataProps }) {
+  const [isDescriptionTruncated, setIsDescriptionTruncated] = useState(true)
+
+  if (!tokenDetailData || !tokenDetailData.description) {
+    return (
+      <AboutContainer>
+        <AboutHeader>
+          <Trans>About</Trans>
+        </AboutHeader>
+        <NoInfoAvailable>
+          <Trans>No token information available</Trans>
+        </NoInfoAvailable>
+        <ResourcesContainer>
+          <Resource name={'Etherscan'} link={`https://etherscan.io/address/${address}`} />
+          <Resource name={'Protocol Info'} link={`https://info.uniswap.org/#/tokens/${address}`} />
+        </ResourcesContainer>
+      </AboutContainer>
+    )
+  } else {
+    const tokenDescriptionArray = tokenDetailData.description.split(' ')
+    const shouldTruncate = tokenDescriptionArray.length > 60
+    const tokenDescriptionTruncated = shouldTruncate ? tokenDescriptionArray.slice(0, 59).join(' ') + '...' : undefined
+
+    return (
+      <AboutContainer>
+        <AboutHeader>
+          <Trans>About</Trans>
+        </AboutHeader>
+        {!shouldTruncate && tokenDetailData.description}
+        {shouldTruncate && isDescriptionTruncated && (
+          <TokenDescriptionContainer>
```

Comment:
this and the next container can be consolidated with ternaries inline

Reply:
sounds good, done